File: osdu-r2/os-python-sdk/osdu_api/model/search/query_request.py
#  Copyright 2020 Amazon
#
#  Licensed under the Apache License, Version 2.0 (the "License");
#  you may not use this file except in compliance with the License.
#  You may obtain a copy of the License at
#
#      http://www.apache.org/licenses/LICENSE-2.0
#
#  Unless required by applicable law or agreed to in writing, software
#  distributed under the License is distributed on an "AS IS" BASIS,
#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
#  See the License for the specific language governing permissions and
#  limitations under the License.

from osdu_api.model.search.sort_query import SortQuery
from osdu_api.model.search.spatial_filter import SpatialFilter

class QueryRequest:

    def __init__(self, kind: str, limit: int, query: str, return_highlighted_fields: bool, 
        returned_fields: list, sort: SortQuery, query_as_owner: bool, spatial_filter: SpatialFilter, 
        from: int, aggregate_by: str):
        self.kind = kind
        self.limit = limit
        self.query = query
        self.return_highlighted_fields = return_highlighted_fields
        self.returned_fields = returned_fields
        self.sort = sort
        self.query_as_owner = query_as_owner
        self.spatial_filter = spatial_filter
        self.from = from
        self.aggregate_by = aggregate_by

File: osdu-r2/os-core-common/src/main/java/org/opengroup/osdu/core/common/model/search/QueryRequest.java
/*
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.opengroup.osdu.core.common.model.search;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.swagger.annotations.ApiModelProperty;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.opengroup.osdu.core.common.SwaggerDoc;
import org.opengroup.osdu.core.common.model.search.validation.ValidOffset;

import javax.validation.constraints.Min;

@Data
@NoArgsConstructor
@AllArgsConstructor
@ValidOffset
public class QueryRequest extends Query {

    @Min(value = 0, message = SwaggerDoc.OFFSET_VALIDATION_MIN_MSG)
    @JsonProperty("offset")
    @ApiModelProperty(value = SwaggerDoc.OFFSET_DESCRIPTION, dataType = "java.lang.Integer", example = "0")
    private int from;

    // aggregation: only make it available in pre demo for now
    @ApiModelProperty(value = SwaggerDoc.AGGREGATEBY_DESCRIPTION, dataType = "java.lang.String", hidden = true)
    private String aggregateBy;

    @Override
    public String toString(){
        return new com.google.gson.Gson().toJson(this);
    }
}
